src/server.ts

```diff
@@ -218,90 +189,87 @@ const getCAmsg = (deps, diagnostics, totalCount): string => {
-        if (metadata != null) {
-            logger.info('cache hit for ' + cacheKey);
-            connection.console.log('cache hit for ' + cacheKey);
-            resolve(metadata);
+/* Fetch Vulnerabilities by component-analysis batch api-call */
+const fetchVulnerabilities = async (reqData) => {
+    try {
+        let url = config.server_url;
+        if (config.three_scale_user_token) {
+            url += `/component-analyses/?user_key=${config.three_scale_user_token}`;
         } else {
-            const part = [ecosystem, name, version].map(v => encodeURIComponent(v)).join('/');
-            const options = {};
-                options['url'] = config.server_url;
-                if(config.three_scale_user_token){
-                    options['url'] += `/component-analyses/${part}?user_key=${config.three_scale_user_token}`;
-                } else{
-                    options['url'] += `/component-analyses/${part}/`;
-                }
-                options['headers'] = {
-                    'Authorization' : 'Bearer ' + config.api_token,
-                };
-            logger.debug('get ' + options['url']);
-            connection.console.log('Scanning ' + part);
-            if(process.env.RECOMMENDER_API_URL){
-                request.get(options, (err, httpResponse, body) => {
-                    if(err){
-                        reject(err);
-                    } else {
-                        if ((httpResponse.statusCode === 200 || httpResponse.statusCode === 202)) {
-                            let response = JSON.parse(body);
-                            logger.debug('response ' + response);
-                            metadataCache[cacheKey] = response;
-                            resolve(response);
-                        } else {
-                            reject(httpResponse.statusCode);
-                        }
-                    }
-                });
-            }
+            url += `/component-analyses`;
+        }
+        const headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer ' + config.api_token,
+            'uuid': config.uuid,
+        };
+        const response = await fetch(url , {
+            method: 'post',
+            body:    JSON.stringify(reqData),
+            headers: headers,
+        });
+         
+        if (response.ok) {
+            const respData = await response.json();
+            return respData;
+        } else {
+            return response.status;
         }
-    });
+    } catch(err) {
+        alert(err);
+    }
 };
+
 /* Total Counts of #Known Security Vulnerability and #Security Advisory */
 class TotalCount 
 {
     vulnerabilityCount: number = 0;
     advisoryCount: number = 0;
 };
 
+/* Runs DiagnosticPileline to consume response and generate Diagnostic[] */
+function runPipeline(response, diagnostics, uri, dependencyMap, totalCount) {
+    response.filter(r => {

```

Comment:
It is suppose to be forEach, not filter.
```suggestion
    response.forEach(r => {

```